frontend/packages/dev-console/src/components/topology/Topology.tsx

```diff
@@ -9,24 +11,64 @@ import {
   BaseEdge,
   VisualizationProvider,
   Model,
+  GraphModel,
+  NodeModel,
   BOTTOM_LAYER,
   GROUPS_LAYER,
   DEFAULT_LAYER,
   TOP_LAYER,
   SelectionEventListener,
   SELECTION_EVENT,
+  NODE_POSITIONED_EVENT,
+  GRAPH_POSITION_CHANGE_EVENT,
 } from '@patternfly/react-topology';
-import { useQueryParams } from '@console/shared';
+import { STORAGE_PREFIX, useQueryParams } from '@console/shared';
 import { useExtensions } from '@console/plugin-sdk';
+import { RootState } from '@console/internal/redux';
 import { isTopologyComponentFactory, TopologyComponentFactory } from '../../extensions/topology';
 import { SHOW_GROUPING_HINT_EVENT, ShowGroupingHintEventListener } from './topology-types';
+import { getTopologyGraphModel, setTopologyGraphModel } from './redux/action';
 import { COLA_LAYOUT, layoutFactory } from './layouts/layoutFactory';
 import { componentFactory } from './components';
 import { odcElementFactory } from './elements';
 import TopologyControlBar from './TopologyControlBar';
 
 import './Topology.scss';
 
+const TOPOLOGY_LAYOUT_LOCAL_STORAGE_KEY = `${STORAGE_PREFIX}/topology-layout`;
+const STORED_NODE_LAYOUT_FIELDS = ['id', 'x', 'y', 'collapsed', 'visible', 'style', 'shape'];
+
+type StoredLayout = {
+  nodes: NodeModel[];
+  layout: string;
+};
+
+const getTopologyLayout = (namespace?: string): StoredLayout => {
+  const storedLayoutsString = localStorage.getItem(TOPOLOGY_LAYOUT_LOCAL_STORAGE_KEY);
+  try {
+    const storedLayouts = JSON.parse(storedLayoutsString);
+    return namespace ? storedLayouts[namespace] : storedLayouts;
+  } catch (e) {
+    return undefined;
+  }
+};
+
+const setTopologyLayout = (namespace: string, nodes: NodeModel[], layout: string) => {
+  const currentStore = getTopologyLayout() || {};
```

Comment:
`getTopologyLayout` returns based on the namespace parameter the StoredLayout or ALL StoredLayouts. I would prefer here two functions. Maybe something like this:

```suggestion
const getStoredData = (): Record<string, StoredLayout> => {
  const storedLayoutsString = localStorage.getItem(TOPOLOGY_LAYOUT_LOCAL_STORAGE_KEY);
  try {
    return JSON.parse(storedLayoutsString) || {};
  } catch (e) {
    return {};
  }
};

const getTopologyLayout = (namespace: string): StoredLayout => {
  return getStoredData()[namespace];
};

const setTopologyLayout = (namespace: string, nodes: NodeModel[], layout: string) => {
  const currentStore = getStoredData();
```

Reply:
@jerolimov Updated, PTAL.